### PR TITLE
architecture refactoring

### PR DESCRIPTION
**Pull request by giulianvivan**
Includes:
- mini-server code merged at "/server"
- creation of a subpackage for "updatehub"
- creation of 2 "cmd"s: updatehub and updatehub-server

Signed-off-by: Giulian Gonçalves Vivan <giulian@ossystems.com.br>

### CODE REVIEW

**Comment by otavio** [comment]
@giulianvivan there are race conditions in the new tests. Please take a look.

File: cmd/updatehub-server/main.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	"github.com/Sirupsen/logrus"
+	"github.com/UpdateHub/updatehub/server"
+	"github.com/spf13/cobra"
+)
+
+func main() {
+	var path string
+
+	cmd := &cobra.Command{
+		Use: "updatehub-localserver [PATH]",
```
Comment (otavio) [inline cmd/updatehub-server/main.go]: Command name must be fixed.

File: cmd/updatehub-server/main.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+		Run: func(cmd *cobra.Command, args []string) {
+			if len(args) == 0 {
+				fmt.Println("You must provide a path")
```
Comment (otavio) [inline cmd/updatehub-server/main.go]: Wouldn't be better to use the logger for those errors?

File: server/daemon.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+	return d, nil
+}
+
+func (d *Daemon) Run() {
+	go func() {
+		for {
+			select {
+			case event := <-d.fswatcher.Events:
+				if event.Op == fsnotify.Remove {
```
Comment (otavio) [inline server/daemon.go]: Using a:
`case event.Op` allows for easier reading.

File: cmd/updatehub-server/main.go (hunk 1 above)
Comment (giulianvivan) [inline cmd/updatehub-server/main.go]: done.

File: cmd/updatehub-server/main.go (hunk 2 above)
Comment (giulianvivan) [inline cmd/updatehub-server/main.go]: done.

File: server/daemon.go (hunk 3 above)
Comment (giulianvivan) [inline server/daemon.go]: I'm not sure if I understood this correctly. The "case" statement is for a "select" and not a "switch".

Comment (otavio) [inline server/daemon.go]: Instead of the if blocks you could use the case.

Comment (giulianvivan) [inline server/daemon.go]: done.

**Pull request merged**